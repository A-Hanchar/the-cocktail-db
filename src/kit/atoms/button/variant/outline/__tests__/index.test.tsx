import {render, screen} from '@testing-library/react';

import {Outline} from '..';

describe('<Button.Outline />', () => {
	it('SHOULD match snapshot', () => {
		const {asFragment} = render(<Outline>Button</Outline>);

		expect(asFragment()).toMatchSnapshot();
	});

	it('SHOULD render component with default testID', () => {
		const {getByTestId} = render(<Outline />);

		expect(getByTestId('button-outline')).toBeInTheDocument();
	});

	it('SHOULD render component with passed testID', () => {
		const testID = 'testID';

		const {getByTestId} = render(<Outline testID={testID} />);

		expect(getByTestId(testID)).toBeInTheDocument();
	});

	it('SHOULD render component with passed className', () => {
		const testClassName = 'test-class-name';

		render(<Outline className={testClassName} />);

		const switchComponent = screen.getByTestId('button-outline');

		expect(switchComponent).toHaveClass(testClassName);
	});
});
