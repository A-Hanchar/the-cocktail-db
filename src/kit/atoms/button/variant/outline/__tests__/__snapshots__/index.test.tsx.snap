// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button.Outline /> SHOULD match snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
    data-test-id="button-outline"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
